dataset/download.py

```diff
@@ -0,0 +1,91 @@
+import git
+from git import RemoteProgress
+from pathlib import Path
+from tqdm import tqdm
+
+
+def _get_dataset_labels_name(name: str) -> str:
+    # Assumes sames labels for different resolutions
+    return "_".join([name, "LABELS"])
+
+
+def _get_dataset_name(name: str, res: int) -> str:
+    return "_".join([name, str(res)])
+
+
+def _setup_image_directory(dir_path: Path) -> None:
+    if dir_path.is_file():
+        RuntimeError(f"File found at {dir_path}, should not exist or be directory.")
+    else:
+        dir_path.mkdir(parents=True, exist_ok=True)
+
+
+# Where to download repositories to
+ROOT = Path("dataset")
+
+# * Add repositories that should be downloaded to this dictionary
+# format for each entry should be:
+# name : (url to repo, branch, msg, setup_func)
+# Use `None` for entries not relevent.
+REPOSITORIES_TO_DOWNLOAD = {
+    _get_dataset_labels_name("FFHQ"): (  # Setup FFHQ labels
+        "https://github.com/DCGM/ffhq-features-dataset.git",
+        "master",
+        None,
+        None,
+    ),
+    _get_dataset_name("FFHQ", 256): (  # Setup FFHQ 256 images
+        None,
+        None,
+        "FFHQ 256 dataset instructions:\n"
+        "1. Download dataset from (must register account on Kaggle):\n"
+        + "https://www.kaggle.com/datasets/xhlulu/flickrfaceshq-dataset-nvidia-resized-256px?resource=download\n"
+        + "2. Unzip and place image files in directory:\n"
+        + f"{ROOT}/"
+        + _get_dataset_name("FFHQ", 256)
+        + "/image",
+        lambda: _setup_image_directory(ROOT / _get_dataset_name("FFHQ", 256) / "image"),
```

Comment:
Suggestion:
```
DATASETS = {
  "FFHQ": {
    "labels": (repo, branch, msg),
    "images": {
      256: (repo, branch, msg),
      1024: ...
    }
  }
}
```

This would omit the need for code in the config. All of that functionality could be moved to `main()`. Also, function definitions could be moved below the dictionary.

Reply:
Yee booi